packages/core/fs/src/NodeFS.js

```diff
@@ -58,7 +58,64 @@ export class NodeFS implements FileSystem {
     : searchNative.findFirstFile;
 
   createWriteStream(filePath: string, options: any): Writable {
-    return fsWriteStreamAtomic(filePath, options);
+    // Make createWriteStream atomic
+    let tmpFilePath = getTempFilePath(filePath);
+    let failed = false;
+
+    const move = async () => {
+      if (!failed) {
+        try {
+          await fs.promises.rename(tmpFilePath, filePath);
+        } catch (e) {
+          // This is adapted from fs-write-stream-atomic. Apparently
+          // Windows doesn't like renaming when the target already exists.
+          if (
+            process.platform === 'win32' &&
+            e.syscall &&
+            e.syscall === 'rename' &&
+            e.code &&
+            e.code === 'EPERM'
+          ) {
+            let [hashTmp, hashTarget] = await Promise.all([
+              hashStream(writeStream.__atomicTmp),
+              hashStream(writeStream.__atomicTarget),
+            ]);
+
+            await this.unlink(writeStream.__atomicTmp);
+
+            if (hashTmp != hashTarget) {
+              throw e;
+            }
+          }
+        }
+      }
+    };
+
+    let writeStream = fs.createWriteStream(tmpFilePath, {
+      ...options,
+      fs: {
```

Comment:
Where does this `fs` option come from? I don't see it documented in Node or graceful-fs.

Reply:
It's documented in Node.js since Node v12. See the `fs` option in the docs for [`fs.createWriteStream`'s options](https://nodejs.org/api/fs.html#fscreatewritestreampath-options).